### PR TITLE
Bugfix/esp spi fix

### PR DESCRIPTION
**Pull request by eren-terzioglu**
## Summary

SPI issues fix for ESP RISC-V based devices.
Created from [this issue](https://github.com/apache/nuttx/issues/12897)

## Impact

ESP32-C3, ESP32-C6 and ESP32-H2

## Testing

Tested with these configs:

```
esp32c3-generic:spi
esp32h2-devkit:spi
esp32c6-devkict:spi
```
Also tested with `CONFIG_ESPRESSIF_SPI2_DMA` option

### CODE REVIEW

File: boards/risc-v/esp32c6/esp32c6-devkitm/configs/spi/defconfig [hunk 1]
```diff
@@ -23,6 +23,7 @@ CONFIG_BUILTIN=y
 CONFIG_DEV_ZERO=y
 CONFIG_ESPRESSIF_ESP32C6=y
 CONFIG_ESPRESSIF_SPI2=y
+CONFIG_ESPRESSIF_SPI2_CSPIN=10
```
Comment (FelipeMdeO) [inline boards/risc-v/esp32c6/esp32c6-devkitm/configs/spi/defconfig:26]: Why CSPIN is = 10 if it isn't exposed in dev kit?

Comment (eren-terzioglu) [reply boards/risc-v/esp32c6/esp32c6-devkitm/configs/spi/defconfig:26]: CS pin for C6 is also main UART TX pin. For that reason it needs to be changed

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Comment by tmedicci** [comment]
> Hello @eren-terzioglu ,
> 
> Now I can see CLK signal but frequency is not ok. Can you please check if frequency is ok? I am testing it using oscilloscope and logic analyzer, in both cases I am facing same issue: Frequency is always near than 400 kHz, using default frequency or not. I mean, default frequency seems not be applied to the HW and -f option is not working. Look My results below:image image image

Thanks for reporting. Let's merge this PR and we'll continue investigating about changing the frequency in another one, right?

**Comment by eren-terzioglu** [comment]
Updated, fyi

**Review by tmedicci — APPROVED** [review]

**Pull request merged**